src/main/kotlin/no/nav/familie/tilbake/oppgave/OppdaterEnhetOppgaveTask.kt

```diff
@@ -0,0 +1,67 @@
+package no.nav.familie.tilbake.oppgave
+
+import no.nav.familie.kontrakter.felles.Behandlingstema
+import no.nav.familie.kontrakter.felles.Tema
+import no.nav.familie.kontrakter.felles.oppgave.FinnOppgaveRequest
+import no.nav.familie.kontrakter.felles.oppgave.Oppgavetype
+import no.nav.familie.kontrakter.felles.tilbakekreving.Ytelsestype
+import no.nav.familie.prosessering.AsyncTaskStep
+import no.nav.familie.prosessering.TaskStepBeskrivelse
+import no.nav.familie.prosessering.domene.Task
+import no.nav.familie.tilbake.behandling.BehandlingRepository
+import no.nav.familie.tilbake.behandling.FagsakRepository
+import no.nav.familie.tilbake.common.repository.findByIdOrThrow
+import no.nav.familie.tilbake.integration.familie.IntegrasjonerClient
+import org.slf4j.LoggerFactory
+import org.springframework.stereotype.Service
+import java.util.UUID
+
+@Service
+@TaskStepBeskrivelse(taskStepType = OppdaterEnhetOppgaveTask.TYPE,
+                     maxAntallFeil = 3,
+                     beskrivelse = "Oppdaterer enhet på oppgave",
+                     triggerTidVedFeilISekunder = 300)
+class OppdaterEnhetOppgaveTask(private val oppgaveService: OppgaveService,
+                               private val integrasjonerClient: IntegrasjonerClient,
+                               private val fagsakRepository: FagsakRepository,
+                               private val behandlingRepository: BehandlingRepository) : AsyncTaskStep {
+
+    private val log = LoggerFactory.getLogger(this::class.java)
+
+    override fun doTask(task: Task) {
+        log.info("OppdaterEnhetOppgaveTask prosesserer med id=${task.id} og metadata ${task.metadata}")
+        val oppgavetype = Oppgavetype.valueOf(task.metadata.getProperty("oppgavetype"))
+        val enhetId = task.metadata.getProperty("enhetId")
+        val beskrivelse = task.metadata.getProperty("beskrivelse")
+        val behandlingId = UUID.fromString(task.payload)
+
+
+        val behandling = behandlingRepository.findByIdOrThrow(behandlingId)
+        val fagsak = fagsakRepository.findByIdOrThrow(behandling.fagsakId)
+
+        val finnOppgaveResponse =
+                integrasjonerClient.finnOppgaver(FinnOppgaveRequest(behandlingstema = Behandlingstema.Tilbakebetaling,
+                                                                    oppgavetype = oppgavetype,
+                                                                    saksreferanse = behandling.eksternBrukId.toString(),
+                                                                    tema = fagsak.ytelsestype.tilTema()))
+        if (finnOppgaveResponse.oppgaver.size > 1) {
+            log.error("er mer enn en åpen oppgave for behandlingen")
```

Comment:
vi bør kaste exception her hvis det finnes mer enn en oppgave eller ikke finnes en oppgave

Reply:
mulig, men vi må uansett inn i gosys for å rydde opp i oppgaver, og da kan man like gjerne bytte enhet der også.